File: src/__fixtures__/sample.go
// Package sample demonstrates Go chunking.
package sample

import "fmt"

const Version = "1.0"

// Greet returns a greeting.
func Greet(name string) string {
	return fmt.Sprintf("Hello, %s!", name)
}

// Animal is a basic animal.
type Animal struct {
	Name string
}

// Speaker can speak.
type Speaker interface {
	Speak() string
}

// Speak implements Speaker for Animal.
func (a Animal) Speak() string {
	return "..."
}

func (a *Animal) SetName(n string) {
	a.Name = n
}

File: src/__fixtures__/sample.py
"""Module docstring."""
import os
import sys

MODULE_CONST = 42


# Helper function with leading comment
def greet(name: str) -> str:
    """Return a greeting."""
    return f"Hello, {name}!"


def add(a: int, b: int) -> int:
    return a + b


class Animal:
    """Base animal class."""

    def __init__(self, name: str) -> None:
        self.name = name

    def speak(self) -> str:
        raise NotImplementedError

    def __repr__(self) -> str:
        return f"Animal({self.name!r})"


class Dog(Animal):
    """A dog."""

    def speak(self) -> str:
        return "Woof!"

    @staticmethod
    def species() -> str:
        return "Canis lupus familiaris"


def standalone_at_end() -> None:
    pass
